WooCommerce/Classes/ViewRelated/Orders/Order Details/OrderDetailsViewController.swift

```diff
@@ -729,6 +729,8 @@ private extension OrderDetailsViewController {
     @objc private func collectPaymentTapped() {
         collectPayment()
 
+        viewModel.markOrderPaymentPending()
```

Comment:
I cannot think or any potential side-effects of this right now, but does it feel off to have to handle this in the client? 🤔

Reply:
I wasn't sure as well. Technically a client triggers the start of the payment process, so in some way, it makes sense to mark the order as pending payment. 🤔

However, you are right. I'll bring this up to the backend devs. The challenge could be that failed receipt functionality is driven by two systems:
- WooPayments: Stripe failure webhook -> turn order failed
- WooCommerce: Order turns failed -> trigger email receipt

When an order is already in a failure state, WooCommerce doesn't catch changes that trigger receipts. Regardless, I'll bring this up. Since there are valid cases corner that wouldn't be supported now without turning the order to pending payment, e.g:

1. Create order
2. Attach customer
3. Make a failed payment
4. Order becomes failed
5. Come back to order
6. Change customer email
7. Make a failed payment again
8. If order wasn't set to pending, a new email would not be sent

Reply:
The web frontend uses `/checkout` endpoint which also [sets the order status to pending when the payment is (re)tried as well](https://github.com/woocommerce/woocommerce/blob/c051a5755c2a15d92c57d6444803afc71a4f6696/plugins/woocommerce/src/StoreApi/Utilities/CheckoutTrait.php#L65). 

There's a comment in the code nearby explaining why these status changes are done:

> // Transition the order to pending, and then completed. This ensures transactional emails fire for pending_to_complete events.

Reply:
TIL!